prediction_market_agent/agents/microchain_agent/prompt_handler.py

```diff
@@ -0,0 +1,34 @@
+from loguru import logger
+from prediction_market_agent_tooling.tools.utils import utcnow
+
+from prediction_market_agent.db.db_storage import DBStorage
+from prediction_market_agent.db.models import Prompt
+
+
+# ToDo - Unify PromptHandler, db_storage and LongTermMemory into 2 classes, one per table.
+class PromptHandler:
+    def __init__(
+        self,
+        session_identifier: str | None = None,
+        sqlalchemy_db_url: str | None = None,
+    ):
+        self.session_identifier = session_identifier
+        self.storage = DBStorage(sqlalchemy_db_url=sqlalchemy_db_url)
+        self.storage._initialize_db()
+
+    def save_prompt(self, prompt: str) -> None:
+        """Save item to storage. Note that score allows many types for easier handling by agent."""
+        if not prompt:
+            logger.info("No prompt to save")
+            return
```

Comment:
Why would there be an empty string?

Reply:
I saw occurances of this due to the prompt lagging that was fixed previously - hence added this safety check.

Reply:
So is no longer needed if the lag is fixed?

Reply:
sure, removing